Snowflake.Data/Core/ArrowResultSet.cs

```diff
@@ -60,8 +44,28 @@ public ArrowResultSet(QueryExecResponseData responseData, SFStatement sfStatemen
                 isClosed = false;
 
                 queryId = responseData.queryId;
+                
+                if (responseData.rowsetBase64.Length > 0)
```

Comment:
let's extract it into a metod.For example something like: _currentChunk = ReadChunk(...)

Reply:
Done